### ISSUE TITLE
Method to trigger a lazysizes initiation?

### ISSUE THREAD

**Issue by petulla** [aFarkas/lazysizes#205]
Hi

Is there a method to load a `lazysizes` classed image -- an init() for a specific image class?

**Comment by aFarkas** [comment]
You have two possibilities:

1. Add the class `lazypreload` to the image. (i.e.: `img.classList.add('lazypreload');`).
2. Call the `lazySizes.loader.unveil` method: (i.e.: `lazySizes.loader.unveil(DOMNode);`).

The 1. option is recommended.

**Comment by petulla** [comment]
Yes and no. 

I've set:

`window.lazySizesConfig.loadMode = 0; `

Basically, I have a lot of elements absolutely positioned to the top of a page and they fade in and out as the user scrolls. So I want to control the lazy load to coincide with fade in of an image. 

It seems that the images are still all loading by default, however, because they're absolutely positioned to the top.

Is there a best practice you recommend here?

**Comment by petulla** [comment]
I will add that setting

`window.lazySizesConfig.loadMode = 0;`

Seems to still lazyload all images.

**Comment by aFarkas** [comment]
`window.lazySizesConfig.loadMode` is only for the time before the onload.

If you don't want to load any image, simply remove the class `lazyload` and as soon as it should be transformed add the class `lazyload` and `lazypreload` or invoke the `lazySizes.loader.unveil(DOMNode);` on the element.

**Comment by petulla** [comment]
I don't think this will work. The images will just load normally, free of lazy loading. I want them to not load at all.

**Comment by petulla** [comment]
So I ended up modifying your library, which applies the class lazyloaded at the right time.

However, all images are loaded immediately on page load.

The images are added in React.js. So I init once the components mount and set init to false before the library loads.

Here is my mod if your library:

isNestedVisible is now like this:
```
var isNestedVisible = function(elem, elemExpand){
			var outerRect;
			var parent = elem;
			var visible = $(elem).parents('.item').css('visibility') != 'hidden';
```

CheckElements is now like this -- with the && on isNestedVisible:

```
if ((eLbottom = rect.bottom) >= elemNegativeExpand &&
						(eLtop = rect.top) <= elvH &&
						(eLright = rect.right) >= elemNegativeExpand * hFac &&
						(eLleft = rect.left) <= eLvW &&
						(eLbottom || eLright || eLleft || eLtop) &&
						((isCompleted && isLoading < 3 && !elemExpandVal && (loadMode < 3 || lowRuns < 4))) && isNestedVisible(lazyloadElems[i], elemExpand)){
							console.log('load')
						unveilElement(lazyloadElems[i]);
						loadedSomething = true;
						if(isLoading > 9){break;}
					} else if(!loadedSomething && isCompleted && !autoLoadElem &&
						isLoading < 4 && lowRuns < 4 && loadMode > 2 &&
						(preloadElems[0] || lazySizesConfig.preloadAfterLoad) &&
						(preloadElems[0] || (!elemExpandVal && ((eLbottom || eLright || eLleft || eLtop) || lazyloadElems[i].getAttribute(lazySizesConfig.sizesAttr) != 'auto')))){
						autoLoadElem = preloadElems[0] || lazyloadElems[i];
					}
```

Any chance you know why my images are loading on page load? They are in the viewport but visible returns false with my modification.

**Comment by aFarkas** [comment]
> Any chance you know why my images are loading on page load?

No as long as I can't see your HTML/CSS, I can only guess.

But I can say, that your changes to lazysizes aren't needed. 

1. Visibility check
Visibility is inherited, which means if an ancestor is hidden, the child is also hidden, unless you set it explicitly to visible again. -> You don't need to traverse up the DOM tree.

2. If you want that lazysizes always checks nested visiblity including visibilty hidden, give your image the `data-expand="auto"` attribute and it should always check.

Please give me a test case, otherwise I can't help you.

**Comment by petulla** [comment]
Ok, so here is the project, live: www.msnbc.com/globeisgone

It doesn't lazy load the images successfully. You can see the dom tree structure with .item as visible/opacity. But the images nonetheless load on load.

**Comment by aFarkas** [comment]
I will look into this tomorrow.

**Comment by aFarkas** [comment]
@petulla 

Do the following two things:

1. add the attribute `data-expand="auto"` to your image elements.
2. Give those images or thier wrapper elements initially the style `visibility: hidden`. Currently they get this style only afterwards with the `.fade-tran-out`, but not initially. The class `div.item` uses the style `visibility: none`, which is invalid.
